### DISCUSSION TITLE
Connection and session establishing speed downgrade

### DISCUSSION THREAD

**Discussion by sssolvez** [WalletConnect/walletconnect-monorepo#2583] [Developer Support]
We have been using walletconnect v1 for a long time and now migrating to walletconnect v2 for a web application (vue).

Use cases are the following:
- Web app via PC browser + smartphone wallet app
- Web app via smartphone browser + (same) smartphone wallet app

When using v1 we were using the walletconnect/web3-provider with walletconnect/sign-client whereas for v2 we started using wagmi/core.

However, we noted an extremely downgrade in terms of speed when connecting and establishing sessions.
These operations were almost instant with v1 whereas they sometimes take even longer than 30 seconds with v2.

We understand this behavior is not the expected one but we wanted to verify if other projects were experiencing similar issues and, if so, how they were addressed.
Also we wanted to ask if we should be expecting same speed regarding the operations mentioned between versions.

Thank you, any feedback is welcomed!

**Comment by finessevanes** [comment]
Hey @sssolvez are you able to provide and example of your implementation?

**Reply by sssolvez** [reply]
@finessevanes 

Hello, recently we changed into using the following:
"@walletconnect/sign-client"
"@walletconnect/universal-provider"
"@web3modal/standalone"

This way we observed better results than with wagmi but we still get long connection times from time to time randomly.
**Slow point is the approval method.**

This is a piece of our init code for example:

```
const provider = await UniversalProvider.init({
      relayUrl: "wss://relay.walletconnect.com/",
      projectId,
    });`

await provider.connect({
      skipPairing: true,
      namespaces: {
        eip155: {
          methods: [
            "eth_sendTransaction",
            "eth_signTransaction",
            "eth_sign",
            "personal_sign",
            "eth_signTypedData",
          ],
          chains: [`eip155:${chainID}`],
          events: ["chainChanged", "accountsChanged"],
          rpcMap: {
            [chainID]: import.meta.env.VITE_PALETTE_JSON_RPC,
          },
        },
      },
    });

const signClient = await SignClient.init({
      projectId,
    });
    const { uri, approval } = await signClient.connect({
      requiredNamespaces: {
        eip155: {
          methods: [
            "eth_sendTransaction",
            "eth_signTransaction",
            "eth_sign",
            "personal_sign",
            "eth_signTypedData",
          ],
          chains: [`eip155:${chainID}`],
          events: ["accountsChanged"],
        },
      },
    });

await web3Modal.openModal({ uri, standaloneChains: [`eip155:${chainID}`] });
    const session = await approval();
```

**Comment by rkovrigin** [comment]
I have the same problems with my set-up.
I use flutter to build a mobile dApp, the downgrade in speed is extreme! 
Have you found any solution to that problem?
